### PR TITLE
[DTM0] EOS-18958: Add ADDB2 instrumentation

### PR DESCRIPTION
**Pull request by mehjoshi**
Signed-off-by: Mehul Joshi <mehul.joshi@seagate.com>

### CODE REVIEW

File: motr/idx.c [hunk 1]
```diff
@@ -162,6 +164,9 @@ static int idx_op_init(struct m0_idx *idx, int opcode,
 		oi->oi_dtx = m0_dtx0_alloc(m0c->m0c_dtms, oi->oi_sm_grp);
 		if (oi->oi_dtx == NULL)
 			return M0_ERR(-ENOMEM);
+		did = m0_sm_id_get(&oi->oi_dtx->tx_dtx->dd_sm);
+		cid = m0_sm_id_get(&op->op_sm);
+		M0_ADDB2_ADD(M0_AVI_CLIENT_TO_DIX, cid, did);
```
Comment (ivan-alekhin) [inline motr/idx.c:169]: client-to-dix? shouldn't it be client-to-dtx?

File: dtm0/fop.c [hunk 2]
```diff
@@ -235,6 +270,11 @@ static void m0_dtm0_send_msg(struct m0_dtm0_service *dtms,
 	req               = m0_fop_data(fop);
 	req->dtr_msg      = msg_type;
 	rc = m0_dtm0_tx_desc_copy(txd, &req->dtr_txr) ?: m0_rpc_post(item);
+
+	phase_sm_id = m0_sm_id_get(&fom->fo_sm_phase);
+	rpc_sm_id   = m0_sm_id_get(&item->ri_sm);
+	M0_ADDB2_ADD(M0_AVI_FOM_TO_TX, phase_sm_id, rpc_sm_id);
```
Comment (ivan-alekhin) [inline dtm0/fop.c:276]: Why do we re-use fom-to-tx here? shouldn't it be something like "fom-to-pmsg"?

**Review by ivan-alekhin — COMMENTED** [review]
@just-now , is it a temporary solution? I mean the cases where we are re-using already defined mappings. Are we going to define our own mappings later on?

**Review by just-now — APPROVED** [review]
+1

File: dtm0/fop.c (hunk 2 above)
Comment (just-now) [reply dtm0/fop.c:276]: Let's leave this mapping as is, today, for new scripts mappings are type-less. Eventually we'll remove all typed mappings from the instrumentation schema.

File: motr/idx.c (hunk 1 above)
Comment (just-now) [reply motr/idx.c:169]: shouldn't, please look at my comment above.

**Pull request merged**